saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
+        self.dep_dict = self.read_dep()
+        self.write_run_info()
 
+        self.number_of_isotopes = len(self.isoname)
         shape = (2, self.steps)
         maxshape = (2, None)
-        self.keff_db = self.f.create_dataset('keff_EOC', shape,
-                                             maxshape=maxshape, chunks=True)
-        self.keff_db_0 = self.f.create_dataset('keff_BOC', shape,
-                                               maxshape=maxshape, chunks=True)
+        self.keff_eoc_db = self.f.create_dataset('keff_EOC', shape,
+                                                 maxshape=maxshape, chunks=True)
+        self.keff_boc_db = self.f.create_dataset('keff_BOC', shape,
+                                                 maxshape=maxshape, chunks=True)
 
         shape = (self.steps + 1, self.number_of_isotopes)
         maxshape = (None, self.number_of_isotopes)
-        self.bu_adens_db_0 = self.f.create_dataset('core adensity before reproc',
-                                                   shape, maxshape=maxshape,
-                                                   chunks=True)
-        self.bu_adens_db_1 = self.f.create_dataset('core adensity after reproc',
-                                                   shape, maxshape=maxshape,
-                                                   chunks=True)
-        self.tank_adens_db = self.f.create_dataset('tank adensity',
+        self.driver_before_db = self.f.create_dataset('driver composition before reproc',
+                                                      shape, maxshape=maxshape,
+                                                      chunks=True)
+        self.driver_after_db = self.f.create_dataset('driver composition after reproc',
+                                                     shape, maxshape=maxshape,
+                                                     chunks=True)
+        self.driver_refill_tank_db = self.f.create_dataset('driver refill tank composition',
+                                                           shape, maxshape=maxshape,
+                                                           chunks=True)
+
+        self.blanket_before_db = self.f.create_dataset('blanket composition before reproc',
+                                                       shape, maxshape=maxshape,
+                                                       chunks=True)
+        self.blanket_after_db = self.f.create_dataset('blanket composition after reproc',
+                                                      shape, maxshape=maxshape,
+                                                      chunks=True)
+        self.blanket_refill_tank_db = self.f.create_dataset('blanket refill tank composition',
+                                                            shape, maxshape=maxshape,
+                                                            chunks=True)
+
+        self.fissile_tank_db = self.f.create_dataset('fissile tank composition', shape, maxshape=maxshape,
```

Comment:
These are all too hard coded. Different reactors are going to have different designs. Leave that up to the user to define. Create a geometry or system class to handle it if necessary. Don't hardcode the names of system components unless they're present in every reactor saltproc will model.